lib/galaxy/tools/parameters/validation.py

```diff
@@ -949,6 +986,7 @@ def validate(self, value, trans=None):
     in_range=InRangeValidator,
     length=LengthValidator,
     metadata=MetadataValidator,
+    dataset_metadata_equal=MetadataEqualsValidator,
```

Comment:
Maybe ``metadata_equal`` for consistency with the above type?

What about stronger typing. I feel like I've been heading this way with tool stuff... if ``value`` is used treat it as a string and use a ``value_json`` attribute to ``json.load`` the value for a typed comparison.

I would make these two changes but you write more tools - if you prefer the way you have it here that is totally fine with me.

Reply:
While thinking about this I would suggest to stick with `dataset_metadata_equal`: All the user (ie tool developer) facing validators are prefixed with `dataset_metadata_`. The `metadata` validator is unlikely to be used by tool developers (its automatically added to each data parameter). So maybe this is an acceptable level of inconsistency...